spring-kafka/src/main/java/org/springframework/kafka/listener/DefaultKafkaConsumerFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.kafka.listener;
+
+import java.util.Properties;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+
+/**
+ * @author Gary Russell
+ *
+ */
+public class DefaultKafkaConsumerFactory<K, V> implements ConsumerFactory<K, V> {
+
+	@Override
+	public Consumer<K, V> createConsumer(Properties configs) {
```

Comment:
I guess that's going to be a more common case outside Spring Integration. I'm mainly thinking of replaying a topic - for example in a KV store or Batch scenario. Normally that'd be done via a Template, but due to the nature of the client it would just result in lots of boilerplate.